src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,176 @@
+import React, {
+  ChangeEvent, useCallback, useEffect, useState,
+} from 'react';
+import debounce from 'lodash/debounce';
+import {
+  useHistory, useLocation,
+} from 'react-router-dom';
+import { getPeople } from '../../api/api';
+import { Person, ServerIPerson } from '../../api/interface';
+import { Loader } from '../Loader';
+import { PersonInfo } from '../PersonInfo';
+import { PersonRow } from '../PersonRow';
+
+import './PeopleTable.scss';
+import {
+  filteringPeoleList,
+  sortingList,
+} from '../../api/helper';
+import { TableHeader } from '../TableHeader';
+
+export const initialList: Person[] = [];
+
+export const PeopleTable: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<boolean>(false);
+  const [foundPerson, setFoundPerson] = useState<Person | ServerIPerson | undefined>(undefined);
+  const [isAddingForm, setIsAddingForm] = useState<boolean>(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then(result => {
+        if (initialList.length === 0) {
+          result.forEach((person: ServerIPerson) => {
+            initialList.push({
+              name: person.name,
+              sex: person.sex,
+              born: person.born,
+              died: person.died,
+              father: person.fatherName === null
+                ? undefined
+                : result.find((father: ServerIPerson) => father.name === person.fatherName),
+              mother: person.motherName === null
+                ? undefined
+                : result.find((mother: ServerIPerson) => mother.name === person.motherName),
+              slug: person.slug,
+            });
+          });
+        }
+
+        setPeople(initialList);
```

Comment:
зачем вообще переменная initialList? передавай результат result.map сразу в setState
```suggestion
        setPeople(result.map(..));
```